hammr/commands/image/image.py

```diff
@@ -423,15 +423,19 @@ def do_publish_without_id(self, template):
             appliance = appliance[0]
             publish_all_builders(self, template, appliance)
 
-    def retrieve_publish_image_with_target_format_builder(self, image, builder):
+    def retrieve_publish_image_with_target_format_builder(self, image, builder, cred_account):
         format_type = image.targetFormat.format.name
         publish_method = getattr(publish_builders, "publish_" + generics_utils.remove_special_chars(format_type), None)
         if publish_method:
-            publish_image = publish_method(builder)
+            if hasattr(cred_account, "keystoneVersion"):
```

Comment:
In order to avoid this reference to Openstack specific account, we could give the account to the publish_method function (but it will require that all publish_method have this parameter even when it is unused):
```suggestion
            publish_image = publish_method(builder, cred_account)
```